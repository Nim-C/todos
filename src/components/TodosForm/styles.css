@media screen and (max-width: 991px) {
  .form-container {
    flex-wrap: wrap;
    flex-direction: column;
    align-items: stretch;
  }

  .todo-name-input {
    margin-right: 0 !important;
  }
}
