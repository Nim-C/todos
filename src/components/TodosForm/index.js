import React, { useCallback, useState } from 'react'
import { useDispatch } from 'react-redux'
import { addTodo } from '../../features/todos/todosSlice'
import classnames from 'classnames'
import './styles.css'

const TodosForm = () => {
  const [name, setName] = useState('')
  const dispatch = useDispatch()

  const handleChangeName = useCallback(e => setName(e.target.value), [])
  const handleAddTodo = useCallback(() => {
    if (name) {
      dispatch(addTodo({ name }))
      setName('')
    }
  }, [dispatch, name])

  return (
    <div
      className={classnames(
        'd-flex',
        'justify-content-between',
        'p-0',
        'col-12',
        'form-container',
        'mb-4'
      )}
    >
      <input
        value={name}
        className={classnames(
          'todo-name-input',
          'form-control',
          'mr-3',
          'my-2'
        )}
        type="text"
        placeholder="Add todoâ€™s..."
        onChange={handleChangeName}
      />
      <button
        type="button"
        className={classnames('btn', 'btn-success', 'my-2')}
        onClick={handleAddTodo}
      >
        Add
      </button>
    </div>
  )
}

export default TodosForm
