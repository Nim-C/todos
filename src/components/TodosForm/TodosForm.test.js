import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import store from '../../store'
import TodosForm from './'

test('renders Todos Form without failing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <TodosForm />
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
