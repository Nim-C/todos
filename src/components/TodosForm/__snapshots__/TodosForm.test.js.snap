// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Todos Form without failing 1`] = `
<div
  className="d-flex justify-content-between p-0 col-12 form-container mb-4"
>
  <input
    className="todo-name-input form-control mr-3 my-2"
    onChange={[Function]}
    placeholder="Add todoâ€™s..."
    type="text"
    value=""
  />
  <button
    className="btn btn-success my-2"
    onClick={[Function]}
    type="button"
  >
    Add
  </button>
</div>
`;
